src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { FILTER_OPTIONS } from '../../utils/constants';
+
+type Props = {
+  todos: Todo[],
+  todosToRender: Todo[]
+  setTodosToRender: (arr: Todo[]) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  todosToRender,
+  setTodosToRender,
+}) => {
+  const [selectedButton, setSelectedButton] = useState('All');
+  const itemsLeft
```

Comment:
Consider moving the itemsLeft calculation into a useCallback or useMemo hook to avoid unnecessary re-renders.